packages/sdk/src/storage/useStorage.ts

```diff
@@ -43,7 +43,7 @@ export const useStorage = <T>(key: string, initialValue: T | (() => T)) => {
       if (data.state === 'initial') {
         const item = (await getItem<T>(key)) ?? data.payload
 
-        if (!cancel) {
+        if (!cancel && JSON.stringify(item) !== JSON.stringify(data.payload)) {
```

Comment:
I think the main case you're trying to optimize here is on the first load, when getItem returns null. If that's the case, why not just save the result of the condition above and check if it had used data.payload as a fallback? I think it's simpler than stringifying everything. 

The other option is that you're trying to optimize for the cases where the initial value (data.payload) is equal to the saved value on the storage, which I think it's not very common and maybe not worth optimizing, but I'm not sure.

In any case, wouldn't that mean that the `state` is inconsistent since it'd be already hydrated?

Reply:
I made it simpler, take a look at it again